src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -8921,6 +9055,31 @@ instruct salI_mem_CL(memory dst, rcx_RegI shift, rFlagsReg cr)
   ins_pipe(ialu_mem_reg);
 %}
 
+instruct salI_rReg_rReg(rRegI dst, rRegI src, rRegI shift)
+%{
+  // This is to match that of the memory variance
+  predicate(VM_Version::supports_bmi2() && !n->in(2)->is_Con());
```

Comment:
Hi, the check is to match the predicates of `rReg_rReg` and `mem_rReg` versions so that the ADLC can correctly mark the latter to be the cisc-spill of the former. And the predicate of `salI_mem_rReg` is to prevent `LShiftI (LoadI mem) imm` from being matched by the variable shift bmi2 rule.

Reply:
I have changed the rule to use `ins_cost` instead for the prevention. Thanks.